pyfstat/injection_parameters.py

```diff
@@ -1,83 +1,256 @@
+                logger.warning(
+                    f"Parameter `{parameter_name}`'s prior was specified as a callable, "
+                    "which is not covered by the current implementation of RNGs "
+                    "and hence will not make use of the specified seed. "
+                    "Please, beware of the new implementation of parameter priors. "
+                    "This will raise an error in the future."
+                )
+                self.priors[parameter_name] = parameter_prior
+                continue
+
+            if not isinstance(parameter_prior, Mapping):
+                # If not dictionary, then return as is
+                self.priors[parameter_name] = lambda val=parameter_prior: val
+                continue
+
+            distribution = next(iter(parameter_prior))
+
+            if distribution in _pyfstat_custom_priors:
+                logger.debug(
+                    f"Setting {distribution} distribution form `_pyfstat_custom_priors` "
+                    f"with kwargs `{parameter_prior[distribution]}` "
+                    f"to parameter {parameter_name}"
+                )
+                custom_function = _pyfstat_custom_priors[distribution]
+                self.priors[parameter_name] = functools.partial(
+                    custom_function,
+                    generator=self._rng,
+                    **parameter_prior[distribution],
+                )
+            elif "stats." in distribution:
+                _, stats_function = distribution.split(".")
+                logger.debug(
+                    f"Setting {stats_function} distribution from `scipy.stats` "
+                    f"with kwargs `{parameter_prior[distribution]}` "
+                    f"to parameter {parameter_name}"
+                )
+                rv_object = getattr(stats, stats_function)(
+                    **parameter_prior[distribution]
+                )
+                rv_object.random_state = self._rng
+                self.priors[parameter_name] = rv_object.rvs
+            else:
+                # FIXME Deprecated, to be removed on a future release.
+                logger.warning(
+                    f"Distribution `{distribution}` not found in "
+                    "`_pyfstat_custom_priors` or `scipy.stats` namespace. "
+                    "Current custom distributions are "
+                    f"`{list(_pyfstat_custom_priors.keys())}`. "
+                    "Please, make sure to follow the proper rules to specify "
+                    "a prior distribution."
+                )
+                logger.warning(
+                    f"Parsing parameter `{parameter_name}` using a deprecated API. "
+                    "This will raise an error in the future."
+                )
+                self.priors[parameter_name] = self._deprecated_prior_parsing(
+                    parameter_prior
+                )
+                continue
+
+    def draw(self) -> dict:
```

Comment:
Would be nice to support `size` argument ->
* need to allow passing it through in parser above
* what should the output be - dict of arrays, or structured array?

Reply:
Structured array should be easier to re-use in many cases (e.g. write to text file), and are easy to convert back to a dict of arrays if needed, so sounds like a good option.
Only downside: need to figure out right syntax for setting the right dtype. (By default numpy would cast everything to floats I think.)